src/lsm/forest.zig

```diff
@@ -915,8 +907,34 @@ fn CompactionScheduleType(comptime Forest: type, comptime Grid: type) type {
             const beats_total = half_bar;
             const beats_done = compaction_beat % half_bar;
             const beats_remaining = beats_total - beats_done;
+
             self.beat_input_size = stdx.div_ceil(self.bar_input_size, beats_remaining);
 
+            // This is akin to a dry run for the actual compaction work that is going to happen
```

Comment:
FWIW, in my head `commence` by itself has exactly this semantics, of doing prep before the actual work starts. And commence happens "in parallel". The originator of this naming is I believe dj, with

```
            inline for (std.meta.fields(Grooves)) |field| {
                @field(forest.grooves, field.name).open_commence(&forest.manifest_log);
            }

            forest.manifest_log.open(manifest_log_open_event, manifest_log_open_callback);

```

which follows the same pattern (we notify _all_ trees that we are beginning the opening procedure. Once we notified all trees, we start feeding them with individual events from the manifest in an interleaved order)